src/script_lua.c

```diff
@@ -980,6 +980,8 @@ static int luaRedisGenericCommand(lua_State *lua, int raise_error) {
     /* Clean up. Command code may have changed argv/argc so we use the
      * argv/argc of the client instead of the local variables. */
     freeLuaRedisArgv(c->argv, &c->argc);
+    c->argv_len_sum = 0;
+    c->argv_len = 0;
```

Comment:
you should be setting these (specifically the argv_len) when building / allocating the c->argv.
you can simply set it to argc.
that's what's dictating if a recalloc is needed in rewriteClientCommandArgument